### ISSUE TITLE
jsdoc/require-jsdoc - publicOnly not working with private methods (using #)

### ISSUE THREAD

**Issue by aws-sde** [gajus/eslint-plugin-jsdoc#1212]
This is similar to https://github.com/gajus/eslint-plugin-jsdoc/issues/1124 but it's using `#` instead of TypeScript's `private` 

## Expected behavior

When `jsdoc/require-jsdoc` is configured with `publicOnly: true` for method definitions,  there should be no error for private methods that lack docs.

## Actual behavior

When `jsdoc/require-jsdoc` is configured with `publicOnly: true` for method definitions, it gives an error for [private](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties) methods (prefixed with `#`) that lack docs.

## ESLint Config

```js
    "jsdoc/require-jsdoc": [
      'error',
      {
        publicOnly: true,
        require: {
          'MethodDefinition': true
        }
      }
    ],
```

## ESLint sample

```js
export class MyClass {
  #myPrivateMethod(): void { }
}
```

## Environment

- Node version:  18.19.1
- ESLint version  8.56.0
- `eslint-plugin-jsdoc` version: 48.2.0
